lib/ttfunk/collection.rb

```diff
@@ -0,0 +1,39 @@
+require 'ttfunk'
+
+module TTFunk
+  class Collection
+    def self.open(path)
+      ::File.open(path, "rb") do |io|
+        yield new(io)
+      end
+    end
+
+    def initialize(io)
+      tag = io.read(4)
+      raise ArgumentError, 'not a TTC file' unless tag == 'ttcf'
+
+      _major, _minor = io.read(4).unpack('n*')
+      count = io.read(4).unpack('N').first
+      @offsets = io.read(count * 4).unpack('N*')
+
+      io.rewind
+      @contents = io.read
+      @cache = []
+    end
+
+    def count
+      @offsets.length
+    end
+
+    def each
+      count.times do |index|
+        yield self[index]
+      end
+      self
+    end
+
+    def [](index)
+      @cache[index] ||= TTFunk::File.new(@contents, @offsets[index])
```

Comment:
I wonder if `@content` could be sliced here so that offset wouldn't need to be passed around.